### ISSUE TITLE
Not Working Bi-Optionally

### ISSUE THREAD

**Issue by tazzy989** [blay09/CraftingTweaks#176]
### Minecraft Version

1.19.x

### Mod Loader

Forge

### Mod Loader Version

Forge 43.1.30

### Mod Version

15.1.0

### Describe the Issue

Tested several ways:
Vanilla server, client has forge, balm, crafting tweaks (nothing else): client CAN connect.
Vanilla client, server has forge, balm, crafting tweaks (nothing else): client cannot connect.
Modded client (excluding crafting tweaks), server has forge, balm, crafting tweaks (nothing else): client cannot connect.
Modded server (excluding crafting tweaks), client has forge, balm, crafting tweaks (nothing else): client cannot connect.
Modded on both sides, with one side excluding crafting tweaks: cannot connect either way. 

Server logs when vanilla client attempts to join, while server has crafting tweaks installed:
[09:40:12] [Netty Server IO #4/ERROR] [ne.mi.ne.NetworkRegistry/NETREGISTRY]: Channels [craftingtweaks:network] rejected vanilla connections
[09:40:12] [Netty Server IO #4/INFO] [ne.mi.se.ServerLifecycleHooks/SERVERHOOKS]: Disconnecting VANILLA connection attempt: This server has mods that require Forge to be installed on the client. Contact your server admin for more details.

Client logs when client has crafting tweaks, but server does not:
[01/10/2022 09:42:06 AM] [Netty Client IO #1/ERROR] Channels [craftingtweaks:network] rejected their server side version number
[01/10/2022 09:42:06 AM] [Netty Client IO #1/ERROR] Terminating connection with server, mismatched mod list

Lmk if you need actual full logs or any other details. Not super familiar with uploading them, my apologies.

### Do you use any performance-enhancing mods (e.g. OptiFine)?

Tested with and without, same result.

**Comment by wchristian** [comment]
Can confirm. Was able to connect to a vanilla server just fine. Then that server installed forge for some server-side mods and since then i get exactly the same error message as above:

```
[12:32:09] [Netty Client IO #0/ERROR]: Channels [craftingtweaks:network] rejected their server side version number during listping
[12:32:09] [Netty Client IO #0/INFO]: Server has additional mods that may be needed on the client: sdlink@
[12:32:09] [Netty Client IO #0/INFO]: Client has mods that are missing on server: [jei, smoothboot, spyglass_improvements, inventoryhud, appleskin, balm, jeresources]
[12:32:09] [Render thread/INFO]: Connecting to alteria.playat.ch, 25565
[12:32:10] [Netty Client IO #1/ERROR]: Channels [craftingtweaks:network] rejected their server side version number
[12:32:10] [Netty Client IO #1/ERROR]: Terminating connection with server, mismatched mod list
[12:36:23] [Render thread/INFO]: Connecting to alteria.playat.ch, 25565
[12:36:24] [Netty Client IO #2/ERROR]: Channels [craftingtweaks:network] rejected their server side version number
[12:36:24] [Netty Client IO #2/ERROR]: Terminating connection with server, mismatched mod list
```

If it helps any, the error seems to come from this function:

https://github.com/MinecraftForge/MinecraftForge/blob/069a94f205fc2fda5a330a2bc68207ab54bf00c9/src/main/java/net/minecraftforge/network/NetworkRegistry.java#L261

**Comment by Talyda** [comment]
Will this fix be in the 1.19.2 version at all? I can see it has gone into 1.19.3 version
